src/dir.rs

```diff
@@ -65,13 +65,24 @@ impl Dir {
 
     /// Returns an iterator of `Result<Entry>` which rewinds when finished.
     pub fn iter(&mut self) -> Iter {
-        Iter(self)
+        Iter {
+            dir: self,
+            should_rewind: true
+        }
+    }
+
+    /// Returns an iterator of `Result<Entry>` which doesn't rewind when finished.
+    pub fn iter_norewind(&mut self) -> Iter {
```

Comment:
What's the expected use case for the non-rewinding Iterator?

Reply:
A performance optimization. There's no need to rewind if we're going to drop the parent `Dir` in a while.

Reply:
I think it's a premature optimization.  `rewinddir` does little more than `lseek`.  If there isn't a demonstrated need for it, I'd rather remove it.

Reply:
Fair enough.